portfolio/src/main/webapp/script.js

```diff
@@ -31,16 +31,77 @@ function addRandomGreeting() {
  * Opens a modal with the given image and caption.
  */
 function openSimpleImageModal(modalId, modalImgId, captionId, closeId, imgSrc, captionText) {  
-  var modal = document.getElementById(modalId);
-  var modalImg = document.getElementById(modalImgId);
-  var caption = document.getElementById(captionId);
+  const modal = document.getElementById(modalId);
+  const modalImg = document.getElementById(modalImgId);
+  const caption = document.getElementById(captionId);
 
   modal.style.display = "block";
   modalImg.src = imgSrc;
-  caption.innerHTML = captionText;
+  caption.innerText = captionText;
 
-  var closeBtn = document.getElementById(closeId);
+  const closeBtn = document.getElementById(closeId);
   closeBtn.onclick = function() {
     modal.style.display = "none";
   };
 }
+
+/**
+ * Advance or Reverse the slide currently showing.
+ */
+function advanceSlides(n) {
+  currIndex += n;
+  currIndex = showSlidePicture(currIndex);
+};
+
+/**
+ * Show the image from the gallery at the given position.
+ */
+function showSlidePicture(n) {
+  const slides = document.getElementsByClassName("slide-pic");
+  const caption = document.getElementById("slide-caption");
+
+  // If not on pics.html document wil not contain any slide elements.
+  if (slides.length === 0) {
+    return n;
+  }
+
+  let newIndex = n;
+  if (n >= slides.length) {
```

Comment:
Not sure of functionality here, but even if N is strictly bigger than slides.length, will you still set it to zero? (or are you trying to do a circular wrap around?

Is it also possible to add unit tests here?

Reply:
To clarify, if n is slides.length + 1, do you need newindex to be 0 or 1 (and is that a valid case?)

Reply:
Yes I noticed that too, after reconsidering what should be the correct functionality, I decided to make it more generic and wrap around by the same magnitude that it is greater/smaller by. So yes it is a circular wrap around, I have now refactored and slightly changed it to jump by two if say the index was two larger than the last item.

Reply:
Added unit tests for these cases also.